src/pylibsshext/channel.pyx

```diff
@@ -105,6 +112,25 @@ cdef class Channel:
     def sendall(self, data):
         return self.write(data)
 
+    def read_bulk_response(self, size=1024, stderr=0, timeout=0.001, retry=5):
+        recv = BytesIO()
+        response = b""
+        while retry:
+            data = self.read_nonblocking(size=size, stderr=stderr)
+            if not data:
+                if timeout:
+                    time.sleep(timeout)
+            else:
+                recv.write(data)
+                offset = recv.tell() - size if recv.tell() > size else 0
+                recv.seek(offset)
+                response += recv.read()
+
+            retry = retry - 1
```

Comment:
There's a better way to do retries — exponential backoff: https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/

Here's a lib that we used for the migration script: https://pypi.org/project/backoff/